cmd/serve.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/skuid/changelog/webhooks/github"
+	"github.com/skuid/spec"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+// serveCmd represents the serve command
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Serve a webhook endpoint for PR validation",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		l, _ := spec.NewStandardLogger()
+		zap.ReplaceGlobals(l)
+		var handler func(http.ResponseWriter, *http.Request)
+
+		switch viper.GetString("provider") {
+		case "github":
+			handler = github.GithubWebhook(viper.GetString("secret"), viper.GetString("token"))
+		default:
+			zap.L().Error(
+				fmt.Sprintf("webhook for provider %s isn't supported", viper.GetString("provider")),
+				zap.String("provider", viper.GetString("provider")),
+			)
+			os.Exit(1)
+		}
+
+		mux := http.NewServeMux()
+		mux.HandleFunc("/webhook", handler)
+		server := &http.Server{
+			Addr:    fmt.Sprintf(":%d", viper.GetInt("port")),
+			Handler: mux,
+		}
+
+		// fmt.Printf("Starting changelog webhook server on port %d\n", viper.GetInt("port"))
+		zap.L().Info("starting changelog webhook server", zap.Int("port", viper.GetInt("port")))
+		if err := server.ListenAndServe(); err != nil {
```

Comment:
Can you add request logging, metrics, healthchecks, and `lifecycle.ShutdownOnTerm()` 

(Don't for get to `import _ "github.com/skuid/spec/metrics"`)